### PR TITLE
[swift] Conform ProtoEnum to ProtoCodable

### PR DESCRIPTION
**Pull request by dnkoutso**
This is an incremental step towards https://github.com/square/wire/pull/2793 which should not have any effect or breaking changes.

### CODE REVIEW

File: wire-runtime-swift/src/test/swift/ProtoEncoderTests.swift [hunk 1]
```diff
@@ -43,4 +43,7 @@ final class ProtoEncoderTests: XCTestCase {
 
         XCTAssertEqual(jsonString, "{}")
     }
+    
+#warning("Test decode bool")
```
Comment (dnkoutso) [inline wire-runtime-swift/src/test/swift/ProtoEncoderTests.swift:47]: - [ ] TODO

File: wire-runtime-swift/src/test/swift/ProtoDecoderTests.swift [hunk 2]
```diff
@@ -36,4 +36,6 @@ final class ProtoDecoderTests: XCTestCase {
         XCTAssertEqual(object2, SimpleOptional2())
     }
 
+    #warning("Test decode bool")
```
Comment (dnkoutso) [inline wire-runtime-swift/src/test/swift/ProtoDecoderTests.swift:39]: - [ ] TODO

File: wire-runtime-swift/src/main/swift/Codable/ProtoEnum.swift [hunk 3]
```diff
@@ -18,7 +18,7 @@ import Foundation
 /// Common protocol that all Wire generated enums conform to
 /// - Note: All ProtoEnums will convert to/from their field and string equivalent when serializing via Codable.
 /// This matches the Proto3 JSON spec: https://developers.google.com/protocol-buffers/docs/proto3#json
-public protocol ProtoEnum : LosslessStringConvertible, Codable {
+public protocol ProtoEnum : LosslessStringConvertible, Codable, ProtoCodable {
```
Comment (lickel) [inline wire-runtime-swift/src/main/swift/Codable/ProtoEnum.swift:21]: This is _mildly_ breaking in that you're forced to recompile, but that seems fine / is desirable

File: wire-runtime-swift/src/main/swift/ProtoCodable/ProtoReader.swift [hunk 4]
```diff
@@ -305,6 +305,7 @@ public final class ProtoReader {
     }
 
     /** Decode a message field */
+    @_disfavoredOverload
```
Comment (lickel) [inline wire-runtime-swift/src/main/swift/ProtoCodable/ProtoReader.swift:308]: This coerces the compiler to prefer the proto enum generic over the proto Codable generic

**Review by lickel — APPROVED** [review]
- [ ] get second approval since we paired on this

File: wire-runtime-swift/src/main/swift/ProtoCodable/ProtoReader.swift (hunk 4 above)
Comment (efirestone) [reply wire-runtime-swift/src/main/swift/ProtoCodable/ProtoReader.swift:307]: neat!

**Review by efirestone — APPROVED** [review]

**Pull request merged**